independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -966,6 +966,13 @@ private RegistrationContext registerSyntheticBeans(List<BeanRegistrar> beanRegis
     }
 
     private void addSyntheticBean(BeanInfo bean) {
+        for (BeanInfo b : beans) {
```

Comment:
Wondering if we should make `beans` a `LinkedHashSet` instead? Not sure how iterating on the bean list for each synthetic bean will slow us down at bootstrap for large number of beans.

Reply:
Hm, we do iterate over all beans quite often and since a typical quarkus apps will have hundreds of beans at most I don't think it's worth the effort...